src/common/libjob/strtab.c

```diff
@@ -0,0 +1,63 @@
+/************************************************************\
+ * Copyright 2022 Lawrence Livermore National Security, LLC
+ * (c.f. AUTHORS, NOTICE.LLNS, COPYING)
+ *
+ * This file is part of the Flux resource manager framework.
+ * For details, see https://github.com/flux-framework.
+ *
+ * SPDX-License-Identifier: LGPL-3.0
+\************************************************************/
+
+#if HAVE_CONFIG_H
+#include "config.h"
+#endif
+#include <flux/core.h>
+
+#include "strtab.h"
+
+static const char *format_entry (struct strtab *entry, const char *fmt)
+{
+    switch (fmt ? *fmt : 0) {
+        case 's':
+            return entry->short_lower;
+        case 'S':
+            return entry->short_upper;
+        case 'l':
+            return entry->long_lower;
+        case 'L':
+        default:
+            return entry->long_upper;
+    }
```

Comment:
curious why default to long upper when there is bad input from the user?  Why not EINVAL or "unknown"?

Reply:
Since these are often used to build printf style arguments, it's inconvenient to have to check for error, so I was just thinking having a default was better.  Also it's consistent with the behavior that was there before when an unknown state was passed.